### PR TITLE
Scrum 48 product management

### PR DESCRIPTION
**Pull request by BeattLau**
Sorry this looks like a huge PR. My previous classes hadn't synced to the new branch and were causing issues so I've updated them. **The new classes are for products which are the ones that really need to be reviewed.**
Still need to work on JWT and security but I will go back to them at a later day as I feel like I'm falling behind trying to complete this part.

### CODE REVIEW

File: src/main/java/com/ecommerce/Config/JwtAuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            UsernamePasswordAuthenticationToken authToken = new UsernamePasswordAuthenticationToken(
+                    username, null, new ArrayList<>());
+            authToken.setDetails(new WebAuthenticationDetailsSource().buildDetails(request));
+            context.setAuthentication(authToken);
+            SecurityContextHolder.setContext(context);
+
+        }
+        filterChain.doFilter(request, response);
+    }
+    private List<String> getRolesByUsername(String username) {
```
Comment (moseseth) [inline src/main/java/com/ecommerce/Config/JwtAuthenticationFilter.java]: getRolesByUsername requires access of user role from the database

File: src/main/java/com/ecommerce/Controller/ProductController.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1")
```
Comment (moseseth) [inline src/main/java/com/ecommerce/Controller/ProductController.java:20]: all product endpoints must start with /products

File: src/main/java/com/ecommerce/Controller/ProductController.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    @GetMapping("/{productId}")
+    public ResponseEntity<Products> getProductById(@PathVariable Long productId) {
+        Products product = productsService.getProductByProductId(String.valueOf(productId));
+        if (product != null) {
+            return ResponseEntity.ok(product);
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
+    @PostMapping("/products/add")
```
Comment (moseseth) [inline src/main/java/com/ecommerce/Controller/ProductController.java]: /add is not required

File: src/main/java/com/ecommerce/Controller/AuthController.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+public class AuthController {
+
+    private final UserServiceImpl userService;
+    private final UserRepository userRepository;
+
+    @PostMapping("/register")
+    public ResponseEntity<CustomUser> registerUser(@RequestBody RegisterRequest registerRequest) {
+        return ResponseEntity.ok(userService.registerUser(registerRequest));
+    }
+    @PostMapping("login")
```
Comment (MenilV) [inline src/main/java/com/ecommerce/Controller/AuthController.java]: shouldn't this be `/login`

File: src/main/java/com/ecommerce/Service/AuthenticationService.java [hunk 5]
```diff
@@ -22,12 +21,12 @@ public class AuthenticationService {
...
     private final PasswordEncoder passwordEncoder;
-    private final JwtUtil jwtUtil;
+    private final JwtServiceImpl jwtServiceImpl;
     private final AuthenticationManager authenticationManager;
-@Builder
+    @Builder
     public AuthenticationResponse register(RegisterRequest request) {
         CustomUser user = CustomUser.builder()
-                .id(1L)
+                .userId(1L)
```
Comment (MenilV) [inline src/main/java/com/ecommerce/Service/AuthenticationService.java]: this looks fishy. should the userId be `1` or dynamically changing?

File: src/main/java/com/ecommerce/Service/ProductsService.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    void deleteProducts(Long productId) throws ProductNotFoundException;
+
+    Products getProductByProductName(String productName);
+    Products getProductByProductId(String productId);
+
+    List<Products> getAllProducts();
+
+    boolean productExists(Long productId);
+
+//    Products addProductsToCart(Products products);
```
Comment (MenilV) [inline src/main/java/com/ecommerce/Service/ProductsService.java:25]: do we need this?

File: src/main/java/com/ecommerce/Service/ProductsServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+ /*   public Products addProductsToCart(Long productId, Long userId) {
+        Products product = productRepository.findById(productId)
+                .orElseThrow(() -> new EntityNotFoundException("Product not found with id: " + productId));
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException("User not found with id: " + userId));
+        user.getShoppingCart().add(product);
+        return product;
+    }*/
+}
```
Comment (MenilV) [inline src/main/java/com/ecommerce/Service/ProductsServiceImpl.java:74]: do we need this?

File: target/classes/com/ecommerce/Repository/RolesRepository.class
Comment (MenilV) [inline target/classes/com/ecommerce/Repository/RolesRepository.class:1]: these `.class` files should be in gitignore, I would think the whole `target` needs to be in gitignore.

File: target/maven-archiver/pom.properties [hunk 8]
```diff
@@ -1,3 +1,7 @@
 artifactId=E-commerceFinalProject
+<<<<<<< HEAD
+groupId=com.
+=======
 groupId=com.example
+>>>>>>> main
```
Comment (MenilV) [inline target/maven-archiver/pom.properties]: there are some conflicts we need to resolve. please be careful as this is very dangerous when conflicts remain in code.

File: target/maven-status/maven-compiler-plugin/compile/default-compile/createdFiles.lst [hunk 9]
```diff
@@ -32,6 +44,14 @@ com/ecommerce/User/Service/UserServiceImpl.class
 com/ecommerce/User/Entity/Permission.class
 com/ecommerce/Jwt/JwtAuthenticationEntryPoint.class
 com/ecommerce/User/Service/MyUserDetailsService.class
+<<<<<<< HEAD
+com/ecommerce/Product/Service/ProductsService.class
```
Comment (MenilV) [inline target/maven-status/maven-compiler-plugin/compile/default-compile/createdFiles.lst]: more conflicts!

File: target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst [hunk 10]
```diff
@@ -8,17 +8,34 @@
...
 /Users/laurabeattie/Downloads/E-commerceFinalProject/src/main/java/com/ecommerce/User/Service/AuthenticationService.java
+/Users/laurabeattie/Downloads/E-commerceFinalProject/src/main/java/com/ecommerce/Product/Service/ProductsService.java
+=======
+/Users/laurabeattie/Downloads/E-commerceFinalProject/src/main/java/com/ecommerce/User/Service/UserServiceImpl.java
+/Users/laurabeattie/Downloads/E-commerceFinalProject/src/main/java/com/ecommerce/User/Service/MyUserDetailsService.java
+/Users/laurabeattie/Downloads/E-commerceFinalProject/src/main/java/com/ecommerce/ExceptionHandler/UserAlreadyExistsException.java
+/Users/laurabeattie/Downloads/E-commerceFinalProject/src/main/java/com/ecommerce/Config/AppConfig.java
+/Users/laurabeattie/Downloads/E-commerceFinalProject/src/main/java/com/ecommerce/User/Entity/RoleToUserForm.java
+/Users/laurabeattie/Downloads/E-commerceFinalProject/src/main/java/com/ecommerce/User/Service/AuthenticationService.java
+>>>>>>> main
```
Comment (MenilV) [inline target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst]: even more conflicts!

**Review by MenilV — CHANGES_REQUESTED** [review]
A lot to check in this one, include a must have conflict resolution.

File: src/main/java/com/ecommerce/Config/JwtAuthenticationFilter.java (hunk 1 above)
Comment (BeattLau) [reply src/main/java/com/ecommerce/Config/JwtAuthenticationFilter.java]: updated this, see commit f0fw3b7

File: src/main/java/com/ecommerce/Controller/AuthController.java (hunk 4 above)
Comment (BeattLau) [reply src/main/java/com/ecommerce/Controller/AuthController.java]: updated this see commit c5167b8

File: src/main/java/com/ecommerce/Controller/ProductController.java (hunk 2 above)
Comment (BeattLau) [reply src/main/java/com/ecommerce/Controller/ProductController.java:20]: updated this, see commit e3f325d

File: src/main/java/com/ecommerce/Controller/ProductController.java (hunk 3 above)
Comment (BeattLau) [reply src/main/java/com/ecommerce/Controller/ProductController.java]: updated this, see commit e3f325d

File: src/main/java/com/ecommerce/Service/AuthenticationService.java (hunk 5 above)
Comment (BeattLau) [reply src/main/java/com/ecommerce/Service/AuthenticationService.java]: updated this, see commit c2b2de2

File: src/main/java/com/ecommerce/Service/ProductsService.java (hunk 6 above)
Comment (BeattLau) [reply src/main/java/com/ecommerce/Service/ProductsService.java:25]: I will when I implement shopping cart!

File: src/main/java/com/ecommerce/Service/ProductsServiceImpl.java (hunk 7 above)
Comment (BeattLau) [reply src/main/java/com/ecommerce/Service/ProductsServiceImpl.java:74]: I will when I implement shopping cart!

File: target/classes/com/ecommerce/Repository/RolesRepository.class
Comment (BeattLau) [reply target/classes/com/ecommerce/Repository/RolesRepository.class:1]: updated this, please see commit 5c1d5a1

File: target/maven-archiver/pom.properties (hunk 8 above)
Comment (BeattLau) [reply target/maven-archiver/pom.properties]: Musie kindly resolved all conflicts

**Review by moseseth — APPROVED** [review]

**Pull request merged**